### PR TITLE
Add zephyr,display chosen property and use it to obtain default display controller in the samples

### PR DESCRIPTION
**Pull request by jfischer-no**
This allows us to make samples that use the display (controller) API more generic and remove ifdefery like
`#define DISPLAY_DEV_NAME DT_LABEL(DT_INST(0, ilitek_ili9340))`.

This patch adds bindings for the display controller integrated in a display, like SSD1306.
We also have devices in the tree where panel and controller are separate entities and
are connected via a defined interface, see mimxrt10xx boards. These are described
differently in devicetree. It may make sense to unify it even if it seems a bit to excessive
for device like SSD1306.
However it is about how the drivers are configured, application that uses display (controller)
API is not affected much since the API provides an interface to retrieve the runtime information,
except it is LVGL adaption layer. It is not possible completely remove options like
LVGL_DISPLAY_DEV_NAME without heavy reworking of LVGL adaption layer
(or complete removal of LVGL from zephyr tree, for which I immediately give +1),
because lvgl_init needs the name of the device, and LVGL_*_RES_MAX options to define
internal buffers.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]
Thanks for these changes, nice cleanup

**Review by MaureenHelm — APPROVED** [review]

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
This needs to be documented in the list of chosen nodes

File: dts/bindings/display/ilitek,ili9xxx-common.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 description: ILI9XXX display controllers common properties.
 
-include: spi-device.yaml
+include: [spi-device.yaml, display-controller.yaml]
```
Comment (mbolivar-nordic) [inline dts/bindings/display/ilitek,ili9xxx-common.yaml:7]: This seems to break bisectability by adding required `height` and `width` properties to the following bindings without updating in-tree users:

- dts/bindings/display/ilitek,ili9340.yaml
- dts/bindings/display/ilitek,ili9488.yaml

The dts/bindings/display/ilitek,ili9341.yaml binding is also affected, but doesn't seem to have any in-tree users.

File: boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts [hunk 2]
```diff
@@ -195,14 +196,16 @@
...
+	ili9340: ili9340@0 {
 		compatible = "ilitek,ili9340";
 		reg = <0>;
 		spi-max-frequency = <32000000>;
 		label = "DISPLAY";
 		reset-gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
 		cmd-data-gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
 		rotation = <270>;
+		width = <240>;
+		height = <320>;
```
Comment (mbolivar-nordic) [inline boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:208]: I think these sorts of changes should be put into the previous patch which added the properties to the binding, for bisectability.

File: samples/drivers/display/src/main.c [hunk 3]
```diff
@@ -211,13 +180,12 @@ void main(void)
...
-
-	display_dev = device_get_binding(DISPLAY_DEV_NAME);
+	LOG_INF("Display sample for %s", DT_LABEL(DT_CHOSEN(zephyr_display)));
 
-	if (display_dev == NULL) {
+	display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+	if (!device_is_ready(display_dev)) {
 		LOG_ERR("Device %s not found. Aborting sample.",
-			DISPLAY_DEV_NAME);
+			DT_LABEL(DT_CHOSEN(zephyr_display)));
```
Comment (mbolivar-nordic) [inline samples/drivers/display/src/main.c:188]: `display_dev->name`. Here and elsewhere in this patch, getting the name from the device struct seems a little cleaner.

File: boards/shields/ls0xx_generic/Kconfig.defconfig [hunk 4]
```diff
@@ -13,10 +13,6 @@ config LS0XX
 
 if LVGL
 
-config LVGL_DISPLAY_DEV_NAME
-	string "Display device name"
-	default "LS0XX"
-
 config LVGL_HOR_RES_MAX
 	int "Horizantal resolution"
 	default 128
```
Comment (mbolivar-nordic) [inline boards/shields/ls0xx_generic/Kconfig.defconfig:18]: Do you also need a separate patch removing these? These values should be coming from devicetree now, right?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks very much for this cleanup. I've got a few more comments now that I had time to do a more detailed review; my earlier -1 was just to keep this from being merged too quickly.

File: drivers/display/ssd16xx.c [hunk 5]
```diff
@@ -636,8 +636,8 @@ static int ssd16xx_controller_init(const struct device *dev)
 
 static int ssd16xx_init(const struct device *dev)
 {
-	const struct ssd16xx_config *config = dev->config;
 	struct ssd16xx_data *driver = dev->data;
+	const struct ssd16xx_config *config = driver->config;
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply drivers/display/ssd16xx.c:640]: not just moving around, `= dev->config;` vs. `= driver->config;`
it is part of https://github.com/zephyrproject-rtos/zephyr/pull/39708

File: dts/bindings/display/ilitek,ili9xxx-common.yaml (hunk 1 above)
Comment (jfischer-no) [reply dts/bindings/display/ilitek,ili9xxx-common.yaml:7]: yes, realized it later, I will do better.

File: boards/shields/ls0xx_generic/Kconfig.defconfig (hunk 4 above)
Comment (jfischer-no) [reply boards/shields/ls0xx_generic/Kconfig.defconfig:18]: Yes, except for mimxrt10xx boards.

**Review by vanwinkeljan — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]
LGTM pending Marti's requested changes.

**Review by brockus-zephyr — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

File: drivers/display/display_dummy.c [hunk 6]
```diff
@@ -4,9 +4,14 @@
...
 
-#include <string.h>
+#define DT_DRV_COMPAT zephyr_dummy_dc
 
+#include <string.h>
 #include <drivers/display.h>
+#include <device.h>
+
+#define DUMMY_DISPLAY_X_RES	DT_INST_PROP(0, width)
+#define DUMMY_DISPLAY_Y_RES	DT_INST_PROP(0, height)
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply drivers/display/display_dummy.c]: I was lazy, reworked both, display_dummy.c and display_sdl.c, to support multi-instance.

File: drivers/display/ssd16xx.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (mbolivar-nordic) [reply drivers/display/ssd16xx.c:640]: Ah, thanks! I missed that.

File: samples/subsys/display/cfb_custom_font/src/main.c [hunk 7]
```diff
@@ -11,23 +11,20 @@
...
 {
-	const struct device *display = DEVICE_DT_GET(DISPLAY_NODE);
+	const struct device *display = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
 	int err;
 
 	if (!device_is_ready(display)) {
 		printk("Display device not ready\n");
 	}
 
+	if (display_set_pixel_format(display, PIXEL_FORMAT_MONO10) != 0) {
```
Comment (mbolivar-nordic) [inline samples/subsys/display/cfb_custom_font/src/main.c:23]: This appears to be an unrelated change for this commit; what's going on here?

Comment (jfischer-no) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: sloppiness, will update after https://github.com/zephyrproject-rtos/zephyr/pull/40876 is merged.

Comment (mbolivar-nordic) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: #40876 doesn't seem to have anything to do with this; is that the right link?

Comment (jfischer-no) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: yes, if I push now, CI will fail again because of that.

**Comment by mbolivar-nordic** [comment]
` samples: crypto: exclude CTR mode for mbedtls `

Did you mean to include this commit in this PR?

**Comment by jfischer-no** [comment]
> `samples: crypto: exclude CTR mode for mbedtls`
> 
> Did you mean to include this commit in this PR?

rather to test what happens,
https://github.com/zephyrproject-rtos/zephyr/pull/40765/checks?check_run_id=4447344836
most of the time I can fix the bugs that CI uncovers myself, but here actions log offers no help at all.

**Comment by mbolivar-nordic** [comment]
> rather to test what happens,

Not really sure if that's a yes or a no :). But  #40876 is merged now, so I'm guessing you plan on making one more update to this PR before we can merge it, right?

**Comment by jfischer-no** [comment]
> > rather to test what happens,
> 
> Not really sure if that's a yes or a no :). But #40876 is merged now, so I'm guessing you plan on making one more update to this PR before we can merge it, right?

Actually, nothing more is planned.  What annoys me is only the unstable CI.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
> nothing more is planned

OK, I see now you did split the pixel format change into its own commit, but you just didn't mention it. LGTM, thanks.

**Comment by jfischer-no** [comment]
rebased to include 474d6866d02d5b74efe9e991c9ed7db5f436ca5b

**Pull request merged**